cmd/minishift/cmd/hostfolder/hostfolder.go

```diff
@@ -0,0 +1,159 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hostfolder
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"os"
+
+	"github.com/docker/machine/libmachine"
+	"github.com/minishift/minishift/pkg/minikube/constants"
+	hostfolderActions "github.com/minishift/minishift/pkg/minishift/hostfolder"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/spf13/cobra"
+)
+
+var (
+	mountAll bool
+)
+
+// HostfolderCmd represents the resource command for host folders
+var HostfolderCmd = &cobra.Command{
+	Use:   "hostfolder SUBCOMMAND [flags]",
+	Short: "Manage and control host folders for use by the OpenShift cluster.",
+	Long:  `Manage and control host folders for use by the OpenShift cluster.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		cmd.Help()
+	},
+}
+
+var hostfolderMountCmd = &cobra.Command{
+	Use:   "mount HOSTFOLDER_NAME",
+	Short: "Mount a host folder to the running cluster",
+	Long:  `Mount a host folder to the running cluster`,
+	Run: func(cmd *cobra.Command, args []string) {
+		api := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
+		defer api.Close()
+		host, _ := api.Load(constants.MachineName)
+
+		var err error = nil
+		if mountAll {
+			err = hostfolderActions.MountHostfolders(host.Driver)
+		} else {
+			if len(args) < 1 {
+				fmt.Fprintln(os.Stderr, "usage: minishift hostfolder mount [HOSTFOLDER_NAME|--all]")
+				atexit.Exit(1)
+			}
+			err = hostfolderActions.Mount(host.Driver, args[0])
+		}
+
+		if err != nil {
+			glog.Errorln(err)
+			atexit.Exit(1)
+		}
+
+	},
+}
+
+// TODO: refactor; lot of repetition with previous command
+var hostfolderUmountCmd = &cobra.Command{
+	Use:   "umount HOSTFOLDER_NAME",
+	Short: "Unmount a host folder from the running cluster",
+	Long:  `Unmount a host folder from the running cluster`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) < 1 {
+			fmt.Fprintln(os.Stderr, "usage: minishift hostfolder umount HOSTFOLDER_NAME")
+			atexit.Exit(1)
+		}
+
+		api := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
+		defer api.Close()
+		host, _ := api.Load(constants.MachineName)
+
+		err := hostfolderActions.Umount(host.Driver, args[0])
+		if err != nil {
+			glog.Errorln(err)
+			atexit.Exit(1)
+		}
+	},
+}
+
+var hostfolderListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List an overview of defined host folders",
+	Long:  `List an overview of defined host folders that can be mounted to a running cluster`,
+	Run: func(cmd *cobra.Command, args []string) {
+		api := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
+		defer api.Close()
+		host, _ := api.Load(constants.MachineName)
+
+		hostfolderActions.List(host.Driver)
+	},
+}
+
+var hostfolderAddCmd = &cobra.Command{
```

Comment:
I just tried this on Windows, but after specifying the name of the mount, it always becomes interactive, right? We might need a way to do this non-interactive. I am thinking for example about the installer. If the installer asks the user for the mount details, it still would need to then somehow apply this.

But I am happy to address this at a later stage, especially since we would have a work around if needed.